Fe profile settings skeleton #25

JAFAJ0 opened the pull request:
### What this PR does (required):
- list PR functionality here

### Screenshots / Videos (front-end only):
![24344](https://user-images.githubusercontent.com/86894074/138627031-2002bc1e-a777-4b1b-b877-a467b9090808.png)

### Any information needed to test this feature (required):
- Just build and run!

### Any issues with the current functionality (optional):
- This is more like a template, to show what the front ends look like.

rajivtitus commented on client/src/pages/ProfileSkeleton/ProfileEditionMock/ProfileEditionMock.tsx:
```diff
@@ -0,0 +1,41 @@
+      <Grid item xs={6} sm={6} md={5} elevation={0} component={Paper} square className={classes.back}>
+        <Box className={classes.authWrapper}>
+          <Box width="100%" maxWidth={450} p={3} alignSelf="center">
+            <ProfileSelection name={'Edit profile'} link={'/profileEditionMock'} selected={true} />
```
```suggestion
            <ChatSideBanner name={'Edit profile'} link={'/profileEditionMock'} selected={true} />
```

Was this supposed to be `ChatSideBanner`? I don't see the `ProfileSelection` component being declared in this PR :)

rajivtitus commented on client/src/components/ProfileSelection/ProfileSelection.tsx:
```diff
@@ -0,0 +1,25 @@
+    <Grid container>
+      <Grid item xs>
+        <Button
+          component={Link}
```
You can look into using react-router's `NavLink`, it has a prop called `activeClassName` which can be used rather than passing down a 'selected' prop to check whether the link has been selected (because currently you've hardcoded a single link as 'true', I presume you will be using a state to keep track of this - which is not necessary) 😄 

Ref: https://reactrouter.com/web/api/NavLink

JAFAJ0 replied:
Using NavLink instead of link

rajivtitus commented on client/src/pages/ProfileSkeleton/ProfileEditionMock/ProfileEditionMock.tsx:
```diff
@@ -0,0 +1,101 @@
+          <Box width="100%" maxWidth={450} p={3} alignSelf="center">
+            <Grid container>
+              <Grid item xs>
+                <Typography className={classes.welcome} component="h1" variant="h5">
```
@JAFAJ0 I would recommend using a Switch/Router here to conditionally render components (please create mock pages for each of the links in the sidebar and test out the feature)

JAFAJ0 replied:
Addressed, you can see the switch router is in main. But will discard this during merge and use another PR to combine real pages instead of mocks

rajivtitus commented:
I've added a couple of comments to support a previous comment so that you can understand the implementation of this feature in a better way. Please have a look and address. Thanks :)

rajivtitus commented on client/src/components/ProfileSelection/ProfileSelection.tsx:
```diff
@@ -0,0 +1,25 @@
+import { Link } from 'react-router-dom';
+import Button from '@material-ui/core/Button';
+
+const ChatSideBanner = (props: { name: string; link: string; selected: boolean }): JSX.Element => {
```
```suggestion
const SidebarLink= (props: { name: string; link: string; selected: boolean }): JSX.Element => {
```

Why don't we give this component a more appropriate name 😄

JAFAJ0 commented:
> I've added a couple of comments to support a previous comment so that you can understand the implementation of this feature in a better way. Please have a look and address. Thanks :)

Addressed

JAFAJ0 commented:
@rajivtitus can I merge this one?

rajivtitus commented:
> @rajivtitus can I merge this one?

Please go ahead and merge it!
